src/swe/physics_swe.c

```diff
@@ -269,25 +269,25 @@ static PetscErrorCode RDyCeedOperatorApply(RDy rdy, PetscReal dt, Vec U_local, V
     // 1. Get the CeedVector associated with the "riemannf" CeedOperatorField
     CeedOperatorField riemannf_field;
     CeedVector        riemannf_ceed;
-    SWESourceOperatorGetRiemannFlux(rdy->ceed_rhs.op_src, &riemannf_field);
+    SWESourceOperatorGetRiemannFlux(rdy->ceed.source_operator, &riemannf_field);
     PetscCallCEED(CeedOperatorFieldGetVector(riemannf_field, &riemannf_ceed));
 
-    PetscScalar *u, *f, *f_dup;
+    PetscScalar *u, *f, *host_fluxes;
     PetscMemType mem_type;
-    CeedVector   u_local_ceed = rdy->ceed_rhs.u_local_ceed;
-    CeedVector   s_ceed       = rdy->ceed_rhs.s_ceed;
+    CeedVector   u_local_ceed = rdy->ceed.u_local;
+    CeedVector   s_ceed       = rdy->ceed.sources;
 
     // 2. Sets the pointer of a CeedVector to a PETSc Vec: u_local_ceed --> U_local
     PetscCall(VecGetArrayAndMemType(U_local, &u, &mem_type));
     PetscCallCEED(CeedVectorSetArray(u_local_ceed, MemTypeP2C(mem_type), CEED_USE_POINTER, u));
 
     // 3. Make a duplicate copy of the F as the values will be used as input for the CeedOperator
     //    corresponding to the source-sink term
-    PetscCall(VecCopy(F, rdy->F_dup));
+    PetscCall(VecCopy(F, rdy->ceed.host_fluxes));
```

Comment:
@jeff-cohere, `rdy->ceed.host_fluxes` should not have included the prefix `host_` because when running on GPUs, this PETSc Vec will be created on the GPU itself as one would use `-dm_vec_type <hip|cuda>`. You can fix this in the next operator refactor PR.